src/fu-util-common.c

```diff
@@ -824,12 +824,8 @@ fu_util_convert_description (const gchar *xml, GError **error)
 		g_set_object (&n, n2);
 	}
 
-	/* remove extra newline */
-	if (str->len > 0)
-		g_string_truncate (str, str->len - 1);
-
 	/* success */
-	return g_string_free (g_steal_pointer (&str), FALSE);
+	return g_strchomp (g_string_free (g_steal_pointer (&str), FALSE));
```

Comment:
I think this is fine, but I do worry about returning midway through an allocated string and then g_free'ing. I think glibc handles this okay, but `fu_common_strstrip(str->str)` might be more safe perhaps. Up to you.

Reply:
Adjusted to use `fu_common_strstrip`, it does seem safer.